src/main/java/org/newrelic/nrjmx/v2/JMXFetcher.java

```diff
@@ -226,16 +229,18 @@ private Set<ObjectInstance> queryMBeans(ObjectName objectName) throws JMXConnect
             }
 
             return result;
-        } catch (ConnectException ce) {
+        } catch (JMXConnectionError je) {
+            throw je;
+        } catch (IOException io) {
```

Comment:
from documentation:
Throws:
IOException – A communication problem occurred when talking to the MBean server.